src/run_env.rs

```diff
@@ -43,6 +48,8 @@ impl RuntimeEnvironment {
             commit_sha: None,
             message: None,
             url: None,
+            collector: format!("rust-{}", COLLECTOR_NAME.to_string()),
```

Comment:
This pattern seems a bit janky; but the rust console kept yelling at me.
I'd prefer `format!("rust-{}", COLLECTOR_NAME.to_string())` to sit in a constant at the top of the file, but I wasn't allowed to change the COLLECTOR_NAME value...

Reply:
That's right because it's declared as a static str.

Are you not able to do:

```
static COLLECTOR_NAME: &str = format!("rust-{}", env!("CARGO_PKG_NAME"));
```

?

Reply:
Nah, I get the below error:
```
static COLLECTOR_NAME: &str = format!("rust-{}", env!("CARGO_PKG_NAME"));
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found struct `std::string::String`
```

Adding a `.to_string()` to the end of the `env!` and changing the expected type to `String` also threw errors which is probably unsurprising.

Reply:
Still getting errors, but I think what I have is fine. It was just triggering my DRY ruby spidey-sense - thank you for the help though!

```
static COLLECTOR_NAME: &str = format!("rust-{}", env!("CARGO_PKG_NAME")).as_str();
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^        - value is dropped here
  |                               |
  |                               the destructor for this type cannot be evaluated in statics

static COLLECTOR_NAME: &str = format!("rust-{}", env!("CARGO_PKG_NAME")).as_str();
  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^---------
  |                               |                                                 |
  |                               |                                                 temporary value is freed at the end of this statement
  |                               creates a temporary which is freed while still in use
  |                               using this value as a static requires that borrow lasts for `'static`
```

Reply:
Yeah, it's a tricky one.  Shouldn't be this hard thanks Rust :)